<launch>
  <!-- Include the first launch file -->
  <include file="/home/tdillon/EM_tracker/src/ascension/launch/demo.launch" />
  <include file="/home/tdillon/usb_workspace/src/usb_cam/launch/usb_cam.launch" />
    
  <!-- the primary node for computation and visualization -->
  <node name="open3d_visualizer" pkg="gating_data_collection" type="mapping_real_time.py" output="screen" />

  <!-- helpers -->
  <node name="binary_image_visualizer" pkg="gating_data_collection" type="binary_image_visualizer.py" output="screen" />
  <node name="rgb_image_visualizer" pkg="gating_data_collection" type="rgb_image_visualizer.py" output="screen" />
  <node name="mapping_parameters_gui" pkg="gating_data_collection" type="mapping_parameters_gui.py" output="screen" />
  <node name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" args="/usb_cam/image_raw" output="screen" />

  <!-- gating signal visualizer, complete this later -->
  <node name="gating_visualizer_white_pump" pkg="gating_data_collection" type="gating_signal_visualizer_white_pump.py" output="screen" />
  
  <!-- Launch rosserial node for Arduino communication -->
  <node name="rosserial_node" pkg="rosserial_python" type="serial_node.py" output="screen">
    <param name="port" value="/dev/ttyACM0"/>  <!-- Specify your Arduino serial port -->
    <!-- <param name="baud" value="57600"/>  Set the baud rate (match it with Arduino) -->
  </node>


</launch>
